### PR TITLE
Eg 42 pokeball animation

### LINKED ISSUE #42: Create a Pokeball Animation
**Is your feature request related to a problem? Please describe.**
There is no sort of "loading" animation when the user types in the Pokemon they want.

**Describe the solution you'd like**
The solution would be to have a "loading" animation by displaying a Pokeball.

### PR DESCRIPTION
**Pull request by BlackCubes**
## Changes
1. Created a Pokeball component.
2. Passed isPokeball states to Pokeball component as props.
3. Created a useEffect with setTimeout and isPokeball states to dynamically render the Pokeball component in Pokeball.js.
4. Passed isVersus state to Search component as props.
5. Expanded the conditional in onSubmit function which decides to render the Pokeball component by checking isVersus and isEnemy.
6. Styled the Pokeball component with animations in Pokeball.css.

## Purpose
The problem was that there was no loading when the user searches for their own pokemon without the versus option of an enemy combatant.

## Approach
This addresses the problem of showing a pokeball animation whenever the user finds their chosen pokemon, and this creates a better, exciting, and engaging UI experience.

Closes #42

### CODE REVIEW

File: pokemon-app/src/components/Pokeball.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from "react";
+import "../styles/Pokeball.css";
+
+const Pokeball = ({ isPokeball, setIsPokeball }) => {
+  useEffect(() => {
+    if (isPokeball) {
+      const timer = setTimeout(() => setIsPokeball(false), 1700);
+      return () => clearTimeout(timer);
+    }
+  }, [isPokeball, setIsPokeball]);
```
Comment (kmausolf) [inline pokemon-app/src/components/Pokeball.js]: Does `setIsPokeball` need to be a dependency here?  How often does this function change?

File: pokemon-app/src/components/Pokeball.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  return !isPokeball ? null : (
```
Comment (kmausolf) [inline pokemon-app/src/components/Pokeball.js:12]: You can simplify this to `return isPokeball && (`

Comment (kmausolf) [reply pokemon-app/src/components/Pokeball.js:12]: I think, at least.

File: pokemon-app/src/components/Pokeball.js (hunk 1 above)
Comment (BlackCubes) [reply pokemon-app/src/components/Pokeball.js]: That was my question too, but my VSCode gave me an yellow error description to include it. Should I delete it, sir?

Comment (kmausolf) [reply pokemon-app/src/components/Pokeball.js]: If you don't expect this prop to be reassigned at any point during execution, you can leave it out.  VSCode isn't perfect with hooks.

**Review by JayDampitan — APPROVED** [review]
Keyframes for days. Cant wait to see it!

**Review by Lap343 — APPROVED** [review]
Saw the animation and it looks great

**Review by Wolftownn — APPROVED** [review]
Looks great Elias!

**Pull request merged**